src/Lean/Elab/SyntheticMVars.lean

```diff
@@ -208,6 +208,94 @@ private def synthesizeUsingDefault : TermElabM Bool := do
       return true
   return false
 
+/-- Translate zero-based indexes (0, 1, 2, ...) to ordinals ("first", "second", "third", ...)-/
+private def toOrdinalString : Nat -> String
+  | 0 => "first"
+  | 1 => "second"
+  | 2 => "third"
+  | 3 => "fourth"
+  | 4 => "fifth"
+  | n => s!"{n+1}th"
```

Comment:
I'm hesitant to put this function in a place that it might get more general use, because it is arguably wrong for, e.g. 21, for it will output "22th" rather than "22nd".

I think it's pragmatically almost ok here because the idea of typeclasses with 22 arguments seems unlikely.

Reply:
Language is fascinating. I added a warning that this practically maxes out at 19, and will welcome any reason to implement the right function.